Automatisk køyring av `lintr` på GitHub

haugedal opened the pull request.

haugedal committed:
Innstillingar for lintr

haugedal committed:
Automatisk køyring av lintr på endra filer på GitHub

Oppretta .yaml- og .gitignore-fil ved å køyra
usethis::use_github_action("lint-changed-files").

statkarl reviewed (changes requested):
Får feil ved lintr-testing:

```
ERROR: a 'NAMESPACE' file is required
* removing ‘/home/runner/work/_temp/Library/rapwhale’
Error: Process completed with exit code 1.
```

Du må legga til eit steg som bygger dokumentasjonen.

haugedal committed:
Dokumenter pakken før han vert bygd

Sidan me ikkje versjonskontrollerer NAMESPACE-fila
må me byggja dokumentasjonen i eit eige steg.

haugedal committed:
Bruk linters_with_defaults()

Function with_defaults was deprecated in lintr version 3.0.0.
Use linters_with_defaults or modify_defaults instead.
